### PR TITLE
WEBUI-1934: handle single-select hierarchical vocabulary in saved search params

### PR DESCRIPTION
**Pull request by swarnadipa-dev**
https://hyland.atlassian.net/browse/WEBUI-1934

### CODE REVIEW

File: elements/search/nuxeo-search-form.js [hunk 1]
```diff
@@ -581,6 +581,15 @@ Polymer({
                     }
                     return output;
                   });
+                } else if (modifyPayload && value && typeof value === 'object' && value.id && value.properties) {
+                  // Single-select hierarchical vocabulary: reconstruct full path (e.g. "bankN1/normalCard")
+                  let output = value.id;
+                  let entry = value;
+                  while (entry && entry.properties && entry.properties.parent) {
+                    output = `${entry.properties.parent.id}`.concat('/', `${output}`);
+                    entry = entry.properties.parent;
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Good catch — addressed in the latest commit (b761db3 / 184aad4).

Both the single-select branch and the existing array branch now extract `parentId` defensively:

```js
const parent = entry.properties.parent;
const parentId = typeof parent === 'string' ? parent : parent.id ?? parent?.properties?.id;
if (!parentId) break;
```

This handles:
- `properties.parent` as a plain string id
- `properties.parent` as an object with `.id`
- `properties.parent` as an object with `.properties.id`
- Stops traversal cleanly (no `undefined/...` paths) when no usable id is found

4 dedicated unit tests added in `test/nuxeo-search-form.test.js` to cover all these edge cases.

[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Thread resolved — defensive `parentId` extraction (`string | parent.id | parent.properties.id`) was added in commit b761db3 and is now also covered by the new test cases for string-parent and `properties.id` parent shapes added in the latest commit.

File: test/nuxeo-search-form.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          label: 'Child Item',
+          parent: {
+            id: 'parentCategory',
+            properties: { label: 'Parent Category' },
+          },
+        },
+      },
+      ecm_fulltext: '',
+      'cvd:contentViewName': 'my_search',
+    };
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply test/nuxeo-search-form.test.js:57]: Fixed in the latest commit — added test cases for both defensive parent shapes: (a) `properties.parent` as a plain string id, and (b) `properties.parent` as an object with `parent.properties.id`. Both are verified in the single-select and multi-select tests.

File: elements/search/nuxeo-search-form.js [hunk 3]
```diff
@@ -803,7 +818,7 @@ Polymer({
       this.isSavedSearch = this._isSavedSearch();
       this.selectedSearch = search;
       const clonedParams = JSON.parse(JSON.stringify(search.params));
-      this.params = this._mutateParams(clonedParams);
+      this.params = this._mutateParams(clonedParams, true);
       this._navigateToResults();
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js:820]: Good point. The array branch now guards on `item.properties` before calling `toHierarchicalPath`: items without `.properties` (e.g. `ecm_collections` selectivity objects `{id, title, text}`) pass through as-is, preserving the `title`/`text` fields needed by `_checkForCollectionTitle` for widget rendering.

File: elements/search/nuxeo-search-form.js [hunk 4]
```diff
@@ -576,11 +576,26 @@ Polymer({
...
+                  let output = value.id;
+                  let entry = value;
+                  while (entry && entry.properties && entry.properties.parent) {
+                    const parent = entry.properties.parent;
+                    const parentId = typeof parent === 'string' ? parent : (parent.id ?? parent?.properties?.id);
+                    if (!parentId) break;
+                    output = `${parentId}`.concat('/', `${output}`);
+                    entry = typeof parent === 'string' ? null : parent;
+                  }
+                  result[param] = output;
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Done — extracted `toHierarchicalPath(entry)` as a shared helper in the latest commit. Both the array branch and the single-select branch now call it, eliminating the duplication.

File: test/nuxeo-search-form.test.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    // With modifyPayload=true (correct caller behaviour)
+    const result = mutate(savedSearchParams, true);
+    expect(result['my:vocabField']).to.deep.equal(['parentCategory1/childItem1', 'parentCategory2/childItem2']);
+
+    // Without modifyPayload (regression guard: objects must NOT be transformed)
+    const resultNoModify = mutate(savedSearchParams, false);
+    expect(resultNoModify['my:vocabField']).to.deep.equal(savedSearchParams['my:vocabField']);
+
+    // Defensive parent shapes in array items (Copilot review comment #1):
+    // (a) properties.parent is a plain string id
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply test/nuxeo-search-form.test.js]: Removed mention of Copilot review comment number.

File: test/nuxeo-search-form.test.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+    const result = mutate(savedSearchParams, true);
+
+    expect(result).to.not.have.property('dc:title');
+    expect(result['my:vocabField']).to.equal('parentCategory/childItem');
+    expect(result.ecm_fulltext).to.equal('');
+    expect(result['cvd:contentViewName']).to.equal('my_search');
+
+    // Defensive parent shapes (Copilot review comment #1):
+    // (a) properties.parent is a plain string id
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply test/nuxeo-search-form.test.js]: Removed mention of Copilot review comment number.

**Comment by swarnadipa-dev** [comment]
recheck

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by AnilKumarVanga — APPROVED** [review]
